scrapers_next/vt/committees.py

```diff
@@ -0,0 +1,89 @@
+    source = URL(
+        "https://legislature.vermont.gov/committee/loadList/2024/",
+        timeout=10,
+    )
+
+    def process_page(self):
+        for each_committee in self.response.json()["data"]:
+            # name
+            name = each_committee["LongName"]
+            # chamber
+            if "house" in name.lower():
+                chamber = "lower"
+            elif "senate" in name.lower():
+                chamber = "upper"
+            else:
+                chamber = "legislature"
+
+            # classification & parent check
+            classification = "committee"
+            parent = None
+            if "subcommittee" in name.lower():
+                classification = "subcommittee"
+                parent = name.lower().split()[-1].replace(")", "")
+
+            # extracting members
+            members_html = each_committee["Members"].strip()
+            members_html = members_html.replace("</br>", "\n")
+            try:
+                doc = fromstring(members_html)
+            except ParserError:
+                logging.warning(f"No membership data found for: {name}")
+                continue
+
+            cleaner = Cleaner(allow_tags=[""], remove_unknown_tags=False)
+            members_list = cleaner.clean_html(doc).text_content()
+
+            name = (
+                name.lower().replace("house", "").replace("senate", "").strip().title()
+            )
+            com = ScrapeCommittee(
+                name=name,
+                chamber=chamber,
+                parent=parent,
+                classification=classification,
+            )
+
+            for member in members_list.splitlines():
+                if not member.strip():
+                    logging.warning(f"No membership data found for: {name}")
+                    continue
+                try:
+                    member = member.lower().replace("rep.", "").replace("sen.", "")
+                    splits = member.split(",")
+                    member_name = splits[0]
+                    role = " ".join(splits[1 : len(splits)]).strip()
+                except IndexError:
+                    member_name = member
+                    role = "Member"
+
+                if (not role) or ("member" in role.lower()):
+                    role = "Member"
+                member_name = member_name.replace(".", "")
+                com.add_member(member_name.strip().title(), role=role.strip().title())
+
+            # additional check if no member found
+            if not com.members:
+                logging.warning(f"No membership data found for: {name}")
+                continue
+            if each_committee.get("WebSiteURL"):
+                com.add_link(each_committee["WebSiteURL"], note="Committee web page")
+            if each_committee.get("StreamingURL"):
+                com.add_link(
+                    each_committee["WebSiteURL"], note="Committee streaming page"
+                )
```

Comment:
This is my fault, but in looking at the data now I see that this actually stores any non-null values for these urls (including empty strings). Could you push up a commit with this solution:
```
url_dict = {
                "WebSiteURL": "web",
                "StreamingURL": "streaming",
                "ExpensesURL": "expenses"
            }
            for url_type, note_term in url_dict.items():
                if each_committee.get(url_type):
                    if len(each_committee[url_type]):
                        com.add_link(
                            each_committee[url_type],
                            note=f"Committee {note_term} page"
                        )
```
Once you've done that, it'll be ready for merge!

Reply:
(I also decided to just consolidate it into a dictionary, and then include the expense url key in case there are ever values for that added in the future)

Reply:
yes sure